pkg/api/mock/matcher/example_test.go

```diff
@@ -0,0 +1,92 @@
+package matcher
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	clouddnsv1 "go.anx.io/go-anxcloud/pkg/apis/clouddns/v1"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.anx.io/go-anxcloud/pkg/api/mock"
+)
+
+// Object is a custom gomega matcher which is used to test for matching objects
+// on the MockAPI instance by comparing the type and provided field names
+func ExampleObject() {
+	defer GinkgoRecover()
+
+	a := mock.NewMockAPI()
+
+	if err := a.Create(context.TODO(), &clouddnsv1.Record{ZoneName: "example.com", Name: "webmail"}); err != nil {
+		log.Fatalf("failed to create DNS record: %s", err)
+	}
+
+	ok := Expect(a.Existing()).To(
+		ContainElement(
+			Object(&clouddnsv1.Record{ZoneName: "example.com"}, "ZoneName"),
```

Comment:
either add a comment like "we'd have to check if it's existing, too" or just the check for that

Reply:
ah scratch that, this is already done by using `a.Existing()` - maybe add a comment clarifying this though ^^